Formula/opencascade.rb

```diff
@@ -0,0 +1,91 @@
+  end
+
+  option "without-opencl", "Build without OpenCL support"
+  option "with-extras", "Install documentation (~17 MB), source files (~113 MB), samples and templates"
+  option "with-test", "Install tests (~55MB)"
+  deprecated_option "with-tests" => "with-test"
+
+  depends_on "cmake" => :build
+  depends_on "freetype"
+  depends_on "freeimage" => :recommended
+  depends_on "tbb"       => :recommended
+  depends_on "gl2ps"     => :recommended
+  depends_on "doxygen"   if build.with? "extras"
+  depends_on :macos      => :snow_leopard
+
+  conflicts_with "oce", :because => "OCE is a fork for patches/improvements/experiments over OpenCascade"
+
+  def install
+    cmake_args = std_cmake_args
+    cmake_args << "-DCMAKE_PREFIX_PATH:PATH=#{HOMEBREW_PREFIX}"
+    cmake_args << "-DCMAKE_INCLUDE_PATH:PATH=#{HOMEBREW_PREFIX}/lib"
+    cmake_args << "-D3RDPARTY_DIR:PATH=#{HOMEBREW_PREFIX}"
+    cmake_args << "-DINSTALL_TESTS:BOOL=ON" if build.with? "tests"
+    cmake_args << "-D3RDPARTY_TBB_DIR:PATH=#{HOMEBREW_PREFIX}" if build.with? "tbb"
+
+    %w[freeimage gl2ps opencl tbb].each do |feature|
+      cmake_args << "-DUSE_#{feature.upcase}:BOOL=ON" if build.with? feature
+    end
+
+    sdk_path = Pathname.new `xcrun --show-sdk-path`.strip
+    cmake_args << "-D3RDPARTY_TCL_DIR:PATH=/usr"
+    cmake_args << "-D3RDPARTY_TCL_INCLUDE_DIR:PATH=#{sdk_path}/usr/include/"
+    cmake_args << "-D3RDPARTY_TK_INCLUDE_DIR:PATH=#{sdk_path}/usr/include/"
+    cmake_args << "-DCMAKE_FRAMEWORK_PATH:PATH=#{HOMEBREW_PREFIX}/Frameworks"
+    cmake_args << "-D3RDPARTY_FREETYPE_INCLUDE_DIR:PATH=#{HOMEBREW_PREFIX}/include/freetype2"
+    opencl_path = Pathname.new "#{sdk_path}/System/Library/Frameworks/OpenCL.framework/Versions/Current"
+    if build.with?("opencl") && opencl_path.exist?
+      cmake_args << "-D3RDPARTY_OPENCL_INCLUDE_DIR:PATH=#{opencl_path}/Headers"
+      cmake_args << "-D3RDPARTY_OPENCL_DLL:FILEPATH=#{opencl_path}/Libraries/libcl2module.dylib"
+    end
+
+    if build.with? "extras"
+      cmake_args << "-DINSTALL_SAMPLES=ON"
+      cmake_args << "-DINSTALL_DOC_Overview:BOOL=ON"
+    end
+
+    mkdir "build" do
+      system "cmake", "..", *cmake_args
+      system "make", "install"
+
+      if build.with? "extras"
+        prefix.install "../src"
+        prefix.install "../adm"
+        share.install_symlink prefix/"adm"
+      else
+        # Some apss expect resoures in legacy ${CASROOT}/src directory
+        cd prefix do
+          ln_s "share/opencascade/resources", "src"
+        end
+      end
+    end
+
+    # add symlinks to be able to compile against OpenCascade
+    loc = "#{prefix}/mac64/clang"
+    bin.install_symlink Dir["#{loc}/bin/*"]
+    lib.install_symlink Dir["#{loc}/lib/*"]
+  end
+
+  def caveats; <<-EOF.undent
+    Some apps will require this enviroment variable:
+      CASROOT=#{opt_prefix}
+    EOF
+  end
+
+  test do
+    ENV["CASROOT"] = opt_prefix
+    "1\n"==`#{bin}/DRAWEXE -c \"pload ALL\"`
```

Comment:
What does that test do?
It will need to be reworked to use `assert_match` or another predicate.

Reply:
This simply runs the DRAWEXE executable and loads the available modules to ensure opencascade built and installed correctly.  I will see if I can morph this test block into a more acceptable homebrew pattern.